### PR TITLE
Observable Preferences J (Preview) and minor refactor

### PR DESCRIPTION
**Pull request by calixtus**
Follow up to #8336 
Preview as a separate tab pref should now have effect without need to restart jabref.
Also moved the search box in the preview panel below the list.
grafik

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
I don't like the idea of adding the search box below the lists. That's not intuitive. A search bar at the top allows the users to quickly browse through the avaiable styles
The current status is also consistent with the journal abbrevation as well

**Comment by calixtus** [comment]
Well, I did not like the search box over the full length of both list, even though the search filter only applies to the available list.
How about a compromise: i'll put the search box above the available list, but only the available list?

**Comment by calixtus** [comment]
grafik

**Comment by calixtus** [comment]
grafik

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 1]
```diff
@@ -2437,66 +2435,65 @@ private void purgeCustomExportFormats(int from) {
 
     @Override
     public PreviewPreferences getPreviewPreferences() {
-        if (this.previewPreferences == null) {
-            updatePreviewPreferences();
+        if (Objects.nonNull(previewPreferences)) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java:2438]: Usage of Objects.nonNull is only useful in predicates/lambdas:
>    This method exists to be used as a Predicate, filter(Objects::nonNull)

**Comment by Siedlerchr** [comment]
I've found some kind of problems with selecting another style.
grafik

Now select an entry and cycle to the next style with F9, 
it stays at the Customized Preview style and the notification "Preview style changed" is triggered multiple times and the style does not get changed

File: src/main/java/org/jabref/preferences/JabRefPreferences.java (hunk 1 above)
Comment (calixtus) [reply src/main/java/org/jabref/preferences/JabRefPreferences.java:2438]: We already used this method in multiple locations in JabRef and everywhere in the preference conversion complex. For consistency in the code I'd like to continue to use it. The api-note, that this method especially exists for use as a Predicate doesn't mean that it couldn't be used for something else too. 😅

File: src/main/java/org/jabref/preferences/JabRefPreferences.java [hunk 2]
```diff
@@ -702,7 +701,6 @@ private JabRefPreferences() {
         // preview
         defaults.put(CYCLE_PREVIEW, "Preview;" + CitationStyle.DEFAULT);
         defaults.put(CYCLE_PREVIEW_POS, 0);
-        defaults.put(PREVIEW_PANEL_HEIGHT, 0.65);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/preferences/JabRefPreferences.java:705]: Why did you delete this?

Comment (calixtus) [reply src/main/java/org/jabref/preferences/JabRefPreferences.java:705]: Was a dead preference option

**Review by Siedlerchr — APPROVED** [review]
Works for me
